python/lsst/ts/observatory/control/auxtel/calsys.py

```diff
@@ -0,0 +1,148 @@
+# This file is part of ts_observatory_control.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["ATCalSys"]
+
+import asyncio
+
+from ..base_group import BaseGroup
+
+
+class ATCalSys(BaseGroup):
+    """ Implement high level ATCalSys functionality.
+
+    Parameters
+    ----------
+    domain: `salobj.Domain`
+        Domain to use of the Remotes. If `None`, create a new domain.
+    electrometer_index : `int`
+        Electrometer index.
+    fiber_spectrograph_index : `int`
+        FiberSpectrograph index.
+    """
+
+    def __init__(
+        self, electrometer_index=1, fiber_spectrograph_index=-1, domain=None, log=None
+    ):
+
+        super().__init__(
+            components=[
+                f"Electrometer:{electrometer_index}",
+                f"ATMonochromator",
+                f"FiberSpectrograph:{fiber_spectrograph_index}",
+            ],
+            domain=domain,
+            log=log,
+        )
+
+    async def setup_monochromator(self, wavelength, entrance_slit, exit_slit, grating):
+        """Setup Monochromator.
+
+        Parameters
+        ----------
+        wavelength : `float`
+            Wavelength in nm.
+        entrance_slit : `float`
+            Size of entrance slit in mm.
+        exit_slit : `float`
+            Size of exist slit in mm.
+        grating : `int`
+            Grating to select.
+
+        """
+
+        await self.atmonochromator.cmd_updateMonochromatorSetup.set_start(
+            wavelength=wavelength,
+            gratingType=grating,
+            fontExitSlitWidth=exit_slit,
+            fontEntranceSlitWidth=entrance_slit,
+            timeout=self.long_timeout,
+        )
+
+    async def electrometer_scan(self, duration):
+        """Perform an electrometer scan for the specified duration and return
+        a large file object event.
```

Comment:
Please consider "the largeFileObjectAvailable message" or "... event message" or "... event data" or....

I usually consider an "event" a topic, not a message, though I can certainly appreciate your viewpoint. I'm hoping that a word or two more would help disambiguate.